src/handlers/upload.rs

```diff
@@ -80,10 +80,63 @@ pub enum UploadError {
     IpfsFetchFailed(String),
 }
 
+/// Generates documentation for a Sway project and uploads it to IPFS.
+/// Returns the IPFS hash of the uploaded documentation tarball, or None if generation fails.
+async fn generate_and_upload_documentation(
+    unpacked_dir: &Path,
+    file_uploader: &FileUploader<'_, impl PinataClient, impl S3Client>,
+) -> Result<String, UploadError> {
+    // Generate documentation
+    tracing::info!("Generating documentation for project");
+    let output = Command::new("forc")
+        .args(["doc", "--path", unpacked_dir.to_str().unwrap()])
+        .current_dir(unpacked_dir)
+        .output()
+        .map_err(|_| UploadError::FailedToCompile)?;
+
+    if !output.status.success() {
+        return Err(UploadError::FailedToCompile);
+    }
+
+    // Upload documentation
+    let doc_dir = unpacked_dir.join("out/doc");
+    if !doc_dir.exists() {
+        return Err(UploadError::FailedToCompile);
+    }
+
+    let docs_tarball_path = unpacked_dir.join("docs.tgz");
+    create_docs_tarball(&doc_dir, &docs_tarball_path)?;
+
+    tracing::info!(
+        "Uploading documentation: {}",
+        docs_tarball_path.to_string_lossy()
+    );
+    let docs_ipfs_hash = file_uploader.upload_file(&docs_tarball_path).await?;
+
+    Ok(docs_ipfs_hash)
+}
+
+/// Creates a compressed tarball of the documentation directory.
+fn create_docs_tarball(docs_dir: &Path, output_path: &Path) -> Result<(), UploadError> {
+    let tar_gz = File::create(output_path).map_err(|_| UploadError::OpenFile)?;
+    let enc = GzEncoder::new(tar_gz, Compression::default());
+    let mut tar = tar::Builder::new(enc);
+
+    tar.append_dir_all(".", docs_dir)
+        .map_err(|_| UploadError::CopyFiles)?;
+    tar.finish().map_err(|_| UploadError::CopyFiles)?;
```

Comment:
nit: additional tracing
```suggestion
    tar.append_dir_all(".", docs_dir)
        .map_err(|e| {
            tracing::error!("Failed to add docs to tarball: {}", e);
            UploadError::CopyFiles
        })?;
    tar.finish().map_err(|e| {
        tracing::error!("Failed to finalize docs tarball: {}", e);
        UploadError::CopyFiles
    })?;
```